### PR TITLE
Fix turbine dep on m4

### PR DESCRIPTION
**Pull request by jmjwozniak**

### CODE REVIEW

File: var/spack/repos/builtin/packages/turbine/package.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ class Turbine(AutotoolsPackage):
     depends_on('autoconf', type='build')
     depends_on('automake', type='build')
     depends_on('libtool', type='build')
+    depends_on('m4', type=('build', 'run'))
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/turbine/package.py]: @jmjwozniak generally `m4` is only a build dependency, do you expect that this package will need it after building?

**Comment by scheibelp** [comment]
@jmjwozniak will you have a change to address https://github.com/spack/spack/pull/10948#discussion_r269280747 soon?

File: var/spack/repos/builtin/packages/turbine/package.py (hunk 1 above)
Comment (jmjwozniak) [reply var/spack/repos/builtin/packages/turbine/package.py]: m4 is a build and run dependency.  We use m4 at run time to generate scheduler job files for the workflows.

**Comment by jmjwozniak** [comment]
@scheibelp just a note to please merge this.  Thanks!

**Comment by scheibelp** [comment]
Thanks for the reminder!

**Pull request merged**